src/object.c

```diff
@@ -1236,6 +1237,13 @@ struct redisMemOverhead *getMemoryOverheadData(void) {
         mh->db[mh->num_dbs].overhead_ht_expires = mem;
         mem_total+=mem;
 
+        if (server.cluster_enabled) {
```

Comment:
need set `overhead_ht_slot_to_keys` to 0 when `server.cluster_enabled` is false, since we use `zrealloc` and added memory is not initialized.

Reply:
i'm still in favor of removing that `if` and the include for cluster.h, and using `dictMetadataSize(c->db->dict)` unconditionally.. this will resolve that bug too.